File: intel-hex-to-bin.java/src/test/java/demo/RecordTest.java
package demo;

import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

/**
 * Unit tests for {@link Record}
 */
public class RecordTest {

    @Test
    public void constructorTest() {
        Row row = new Row(":021234005678EA");
        Record record = new Record(row);

        assertEquals(record.getByteCount(), (short) 0x02);
        assertEquals(record.getAddress(), 0x1234);
        assertEquals(record.getRecordType(), RecordType.of((short) 0x00));
        assertEquals(record.getData(), new short[]{0x56, 0x78});
        assertEquals(record.getChecksum(), (short) 0xEA);
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void checksumFailureConstructorTest() {
        Row row = new Row(":02123400567800");
        new Record(row);
    }

}

File: intel-hex-to-bin.kotlin/src/test/kotlin/demo/RecordTest.kt
package demo

import org.testng.Assert.assertEquals
import org.testng.annotations.Test

/**
 * Unit tests for [Record]
 */
class RecordTest {

    @Test
    fun constructorTest() {
        val row = Row(":021234005678EA")
        val record = Record(row)

        assertEquals(record.byteCount, 0x02.toShort())
        assertEquals(record.address, 0x1234)
        assertEquals(record.recordType, 0x00.toShort().toRecordType())
        assertEquals(record.data, shortArrayOf(0x56, 0x78))
        assertEquals(record.checksum, 0xEA.toShort())
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun constructorChecksumFailureTest() {
        val row = Row(":02123400567800")
        Record(row)
    }

}
